dll/win32/browseui/regtreeoptions.cpp

```diff
@@ -1,59 +1,565 @@
+    hr = GetRegString(hKey, L"RegPath", SubKey, _countof(SubKey));
+    if (FAILED(hr))
+        return hr;
+    hr = GetRegString(hKey, L"ValueName", Name, _countof(Name));
+    if (FAILED(hr))
+        return hr;
+    DWORD Err, cb;
+    if (Set)
+        Err = RegCreateKeyEx(hValueKey, SubKey, 0, NULL, 0, KEY_READ | KEY_SET_VALUE, NULL, &hValueKey, NULL);
+    else
+        Err = RegOpenKeyExW(hValueKey, SubKey, 0, KEY_READ, &hValueKey);
+    if (Err)
+        return HRESULT_FROM_WIN32(Err);
+
+    if (Set && (HasMask || HasOffset))
+    {
+        Err = SHRegGetValueW(hValueKey, NULL, Name, SRRF_RT_ANY, NULL, NULL, &cb);
+        hr = HResultFromWin32(Err);
+        if (Err == ERROR_FILE_NOT_FOUND)
+        {
+            DWORD DefaultData = 0;
+            cb = sizeof(DefaultData);
+            hr = GetDefaultValue(hKey, Type, &DefaultData, cb);
+            if (SUCCEEDED(hr))
+            {
+                Err = SHSetValueW(hValueKey, NULL, Name, Type, &DefaultData, cb);
+                hr = HResultFromWin32(Err);
+            }
+        }
+
+        if (SUCCEEDED(hr))
+        {
+            hr = HRESULT_FROM_WIN32(ERROR_OUTOFMEMORY);
+            cb = max(cb, (Offset + 1) * sizeof(DWORD));
+            if (DWORD *pBigData = (DWORD*)LocalAlloc(LPTR, cb)) // LPTR to zero extended data
+            {
+                Err = SHRegGetValueW(hValueKey, NULL, Name, SRRF_RT_ANY, &Type, pBigData, &cb);
+                hr = HResultFromWin32(Err);
+                Size = cb;
+                if (SUCCEEDED(hr))
+                {
+                    pBigData[Offset] &= ~Mask;
+                    pBigData[Offset] |= *((DWORD*)Data);
+                    Err = SHSetValueW(hValueKey, NULL, Name, Type, pBigData, cb);
+                    hr = HResultFromWin32(Err);
+                }
+                LocalFree(pBigData);
+            }
+        }
+    }
+    else if (Set)
+    {
+        hr = HResultFromWin32(SHSetValueW(hValueKey, NULL, Name, Type, Data, Size));
+    }
+    else
+    {
+        if (Offset && HasOffset)
+        {
+            Size = sizeof(DWORD);
+            Err = SHRegGetValueW(hValueKey, NULL, Name, SRRF_RT_ANY, NULL, NULL, &cb);
+            if (Err == ERROR_SUCCESS)
+            {
+                Err = ERROR_OUTOFMEMORY;
+                if (DWORD *pBigData = (DWORD*)LocalAlloc(LPTR, cb))
+                {
+                    Err = SHRegGetValueW(hValueKey, NULL, Name, SRRF_RT_ANY, &Type, pBigData, &cb);
+                    if (Offset < cb / sizeof(DWORD) && Err == ERROR_SUCCESS)
+                        *((DWORD*)Data) = pBigData[Offset];
+                    else
+                        *((DWORD*)Data) = 0; // Value not present or offset too large
+                    LocalFree(pBigData);
+                }
+            }
+            hr = HResultFromWin32(Err);
+        }
+        else
+        {
+            hr = HResultFromWin32(SHRegGetValueW(hValueKey, NULL, Name, SRRF_RT_ANY, &Type, Data, &Size));
```

Comment:
```suggestion
            Err = SHRegGetValueW(hValueKey, NULL, Name, SRRF_RT_ANY, &Type, Data, &Size);
            hr = HResultFromWin32(Err);
```

Reply:
Now that I'm looking again at these hr, I wonder whether it could be possible to just initialize the Err, and factor out the `hr = ...` to be placed just after the end of some of these if-else blocks.